src/Cutscenes.ts

```diff
@@ -1396,6 +1396,130 @@ export class Cutscenes<TEightBittr extends FullScreenPokemon> extends EightBittr
+        const defenderName: string = args.defenderName;
+        const attackerName: string = args.attackerName;
+        const defender: IThing = this.EightBitter.BattleMover.getThing(defenderName) as IThing;
+        const attacker: IThing = this.EightBitter.BattleMover.getThing(attackerName) as IThing;
+        const dt: number = 1;
+        const size: number = this.EightBitter.unitsize;
+        const direction: number = defenderName === "opponent" ? -1 : 1;
+        const lineArray: IThing [] = [];
+        const menu: IMenu = this.EightBitter.MenuGrapher.getMenu("BattleDisplayInitial") as IMenu;
+        const icicle1: IThing = this.EightBitter.ObjectMaker.make("IceBeamIcicles1");
+        const icicle2: IThing = this.EightBitter.ObjectMaker.make("IceBeamIcicles2");
+        const icicle3: IThing = this.EightBitter.ObjectMaker.make("IceBeamIcicles3");
+        const icicle4: IThing = this.EightBitter.ObjectMaker.make("IceBeamIcicles4");
+        let startX: number;
+        let startY: number;
+        let endX: number;
+        let endY: number;
+        let differenceX: number;
+        let differenceY: number;
+        let iceBall: IThing;
+        if (direction === -1) {
+            iceBall = this.EightBitter.ObjectMaker.make("IceBeamCircle");
+        } else {
+            iceBall = this.EightBitter.ObjectMaker.make("IceBeamCircleReverse");
+        }
+        if (direction === -1) {
+            startX = menu.left + attacker.width * 1.5 * this.EightBitter.unitsize;
+            startY = menu.bottom - attacker.height * 1.5 * this.EightBitter.unitsize;
+            endX = menu.right - defender.width * this.EightBitter.unitsize;
+            endY = menu.top + (defender.height / 2) * this.EightBitter.unitsize;
+        } else {
+            startX = menu.right - defender.width * this.EightBitter.unitsize;
+            startY = menu.top + (defender.height / 2) * this.EightBitter.unitsize;
+            endX = menu.left + attacker.width * this.EightBitter.unitsize;
+            endY =  menu.bottom - attacker.height * this.EightBitter.unitsize;
+        }
+        differenceX = (endX - startX) / 32;
+        differenceY = (endY - startY) / 32;
+        this.EightBitter.things.add(iceBall, startX, startY);
+        this.EightBitter.TimeHandler.addEventInterval (
+            (): void => {
+                const left: number = direction === -1 ? iceBall.left : iceBall.right - 3 * this.EightBitter.unitsize;
+                const top: number =  iceBall.bottom - 3 * this.EightBitter.unitsize;
+                this.EightBitter.TimeHandler.addEvent(
+                    (): void => this.EightBitter.physics.shiftHoriz(iceBall, differenceX),
+                    dt);
+                this.EightBitter.TimeHandler.addEvent(
+                    (): void => this.EightBitter.physics.shiftVert(iceBall, differenceY),
+                    dt);
```

Comment:
This feels weird. Why are you delaying the death of the ice balls by 1? Does it make a visual difference?